### ISSUE TITLE
Sometime I get running error, maybe it is a bug

### ISSUE THREAD

**Issue by lixinsu** [google-research/bert#211]
https://github.com/google-research/bert/blob/4a47cc2da23dcb4ab1bf5c08085910b6fd94a4cf/run_squad.py#L865-L885
This may cause `best_non_null_entry` is None when `FLAGS.version_2_with_negative` is True, and I think `if not nbest` should be moved up.

**Comment by ymcui** [comment]
Yes, sometimes the `best_non_null_entry` can be `null` and breaks the prediction process. It would be better to add a `if/else` to avoid such unexpected behavior.
